File: backend/src/main/java/com/quickcart/entity/Order.java
package com.quickcart.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_number", nullable = false, unique = true)
    private String orderNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private Customer customer;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal subtotal;

    @Column(name = "delivery_fee", nullable = false, precision = 10, scale = 2)
    private BigDecimal deliveryFee;

    @Column(precision = 10, scale = 2)
    @Builder.Default
    private BigDecimal discount = BigDecimal.ZERO;

    @Column(name = "total_amount", nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_method", nullable = false)
    @Builder.Default
    private PaymentMethod paymentMethod = PaymentMethod.COD;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_status", nullable = false)
    @Builder.Default
    private PaymentStatus paymentStatus = PaymentStatus.PENDING;

    @Enumerated(EnumType.STRING)
    @Column(name = "order_status", nullable = false)
    @Builder.Default
    private OrderStatus orderStatus = OrderStatus.CREATED;

    @Column(name = "delivery_address_snapshot", columnDefinition = "TEXT", nullable = false)
    private String deliveryAddressSnapshot;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> orderItems;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum PaymentMethod {
        COD, ONLINE
    }

    public enum PaymentStatus {
        PENDING, COLLECTED, FAILED
    }

    public enum OrderStatus {
        CREATED,
        PARTNER_SEARCHING,
        PARTNER_ASSIGNED,
        PARTNER_ACCEPTED,
        PICKING,
        PACKED,
        DELIVERY_SEARCHING,
        DELIVERY_ASSIGNED,
        PICKED_UP,
        OUT_FOR_DELIVERY,
        DELIVERED,
        CANCELLED,
        OUT_OF_STOCK,
        DELIVERY_FAILED
    }
}

File: lib/models/order.dart
import 'cart_item.dart';
enum OrderStatus { created, partnerSearching, partnerAssigned, partnerAccepted, picking, packed, deliverySearching, deliveryAssigned, pickedUp, outForDelivery, delivered, cancelled }
extension OrderStatusX on OrderStatus { String get label => switch(this){OrderStatus.created=>'Order placed',OrderStatus.partnerSearching=>'Finding nearby store',OrderStatus.partnerAssigned=>'Store assigned',OrderStatus.partnerAccepted=>'Store accepted',OrderStatus.picking=>'Packing your order',OrderStatus.packed=>'Packed',OrderStatus.deliverySearching=>'Finding delivery partner',OrderStatus.deliveryAssigned=>'Delivery partner assigned',OrderStatus.pickedUp=>'Picked up',OrderStatus.outForDelivery=>'Out for delivery',OrderStatus.delivered=>'Delivered',OrderStatus.cancelled=>'Cancelled'}; }
class CustomerOrder {
 final String id; final DateTime createdAt; final List<CartItem> items; final double subtotal,deliveryFee,discount,total; final String address,paymentMethod; OrderStatus status;
 CustomerOrder({required this.id,required this.createdAt,required this.items,required this.subtotal,required this.deliveryFee,required this.discount,required this.total,required this.address,required this.paymentMethod,this.status=OrderStatus.created});
}
